_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -988,6 +988,11 @@ mfxStatus VideoDECODEAV1::SubmitFrame(mfxBitstream* bs, mfxFrameSurface1* surfac
                 return MFX_ERR_NONE;
             }
 
+            if (sts == MFX_ERR_NONE && umcRes != UMC::UMC_OK)
+            {
+                sts = MFX_ERR_UNKNOWN;
```

Comment:
maybe i missed smth but here u just set error status and continue `for` loop, so i see no side effects

Reply:
Exit from loop depends on this variable:
https://github.com/Intel-Media-SDK/MediaSDK/blob/6b6cc790c08426b4111094fa311dd34e2e4541fb/_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp#L897
It is equal to `umcRes` (or it can be set to `UMC::UMC_ERR_NOT_ENOUGH_BUFFER` during error handling).
Anyway, if it is not `UMC_OK` (so `sts` is not `MFX_ERR_NONE`) we are going to exit the loop. According to this:
https://github.com/Intel-Media-SDK/MediaSDK/blob/6b6cc790c08426b4111094fa311dd34e2e4541fb/_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp#L996-L997

Reply:
@dmitry-gurulev : was your question answered?

Reply:
Well... I looked for the second time into the change... I think @dmitry-gurulev is actually right... This is incorrect fix.

@aobolensk : can you, please, rewrite this whole code block. We need a clear sequence like:

```
if (umcRes == UMC::UMC_OK) {do something}
if (umcRes == UMC::UMC_ERR_INVALID_STREAM) {do something}
else if (UMC::UMC_NTF_NEW_RESOLUTION) {do something}
else { do something}  // <<<<< your are actually should add this.
```

or you can do the same with switch if this will make more sense....

Reply:
Replaced it with switch. Errors and warnings that change umcRes itself are handled separately.